src/services/mongoose/accounts-ips.ts

```diff
@@ -40,33 +46,54 @@ export const AccountsIpRepository = (): IAccountsIPsRepository => {
     }
   }
 
-  const findById = async (
-    accountId: AccountId,
-  ): Promise<AccountIPs | RepositoryError> => {
+  const findByAccountIdAndIp = async ({
+    accountId,
+    ip,
+  }: FindByAccountIdAndIpInput): Promise<AccountIP | RepositoryError> => {
     try {
-      const result = await Account.findOne(
-        { _id: toObjectId<AccountId>(accountId) },
-        { lastIPs: 1 },
-      )
+      const result = await AccountIps.findOne({
+        _accountId: toObjectId<AccountId>(accountId),
+        ip,
+      })
       if (!result) {
-        return new CouldNotFindUserFromIdError(accountId)
+        return new CouldNotFindAccountIpError(accountId)
       }
 
-      return userIPsFromRaw(result)
+      return accountIPFromRaw(result)
     } catch (err) {
       return parseRepositoryError(err)
     }
   }
 
+  const findLastByAccountId = async (
+    accountId: AccountId,
+  ): Promise<AccountIP | RepositoryError> => {
+    try {
+      const result = await AccountIps.findOne({
+        _accountId: toObjectId<AccountId>(accountId),
+      }).sort({ lastConnection: -1 })
```

Comment:
if we will find only by account  maybe we should add an additional index with lastConnection

Reply:
have been thinking about it. it's currently only used by earn so don't think that's necessary